### PR TITLE
feat: adding annotation for PKCE in OAuth 2

### PR DESCRIPTION
**Pull request by jarrodek**
Adds an annotation supported by API Console to handle OAuth 2 authorization for the `code` grant type with PKCE extension as defined in [RFC 7636](https://tools.ietf.org/html/rfc7636).

### CODE REVIEW

**Comment by jarrodek** [comment]
@jstoiko Could you, please, approve this PR?

File: annotations/security-schemes/README.md [hunk 1]
```diff
@@ -11,14 +11,26 @@
...
+
 - oauth-2-custom-settings.raml
 
 and the following example:
+
 - oauth-2-custom-settings-example.raml
 
 ### Which tools support this
 
 - [mulesoft/api-console (v5.0.x)](https://github.com/mulesoft/api-console)
```
Comment (jstoiko) [inline annotations/security-schemes/README.md]: Do you want to bump the version here or maybe say `> v5.0`?

File: annotations/security-schemes/README.md [hunk 2]
```diff
@@ -11,14 +11,26 @@
...
+
+## `pkce`
+
+### Purpose
+
+Defines annotation that adds support for RFC 7636: Proof Key for Code Exchange.
+
+### Which tools support this
+
+- [mulesoft/api-console (v6.x.x)](https://github.com/mulesoft/api-console)
```
Comment (jstoiko) [inline annotations/security-schemes/README.md]: `> v6.0`?

File: annotations/security-schemes/README.md [hunk 3]
```diff
@@ -11,14 +11,26 @@
...
 
 and the following example:
+
 - oauth-2-custom-settings-example.raml
 
 ### Which tools support this
 
 - [mulesoft/api-console (v5.0.x)](https://github.com/mulesoft/api-console)
+
+## `pkce`
```
Comment (jstoiko) [inline annotations/security-schemes/README.md:28]: There is no `What's included` subsection for pkce, is it intended? Might be worth adding one, with link to the respective files...

File: annotations/security-schemes/README.md (hunk 1 above)
Comment (jarrodek) [reply annotations/security-schemes/README.md]: Nah, it's still valid for 5.x, even though we don't support it.

File: annotations/security-schemes/README.md (hunk 2 above)
Comment (jarrodek) [reply annotations/security-schemes/README.md]: Yes, will fix that.

File: annotations/security-schemes/README.md (hunk 3 above)
Comment (jarrodek) [reply annotations/security-schemes/README.md:28]: My bad, I'll add it in a minute,

File: annotations/security-schemes/README.md (hunk 1 above)
Comment (jarrodek) [reply annotations/security-schemes/README.md]: Ah, I know what you mean. I'll fix that.

File: annotations/security-schemes/README.md (hunk 2 above)
Comment (jarrodek) [reply annotations/security-schemes/README.md]: Fixed

File: annotations/security-schemes/README.md (hunk 1 above)
Comment (jarrodek) [reply annotations/security-schemes/README.md]: Fixed

File: annotations/security-schemes/README.md [hunk 4]
```diff
@@ -11,14 +11,37 @@
...
 
 This directory contains the following RAML `AnnotationTypeDeclaration` fragment:
+
 - oauth-2-custom-settings.raml
 
 and the following example:
+
 - oauth-2-custom-settings-example.raml
 
 ### Which tools support this
```
Comment (jstoiko) [inline annotations/security-schemes/README.md]: I think this was meant to be a question, it's missing the `?` at the end.

File: annotations/security-schemes/README.md [hunk 5]
```diff
@@ -11,14 +11,37 @@
 
 (see full [description](oauth-2-custom-settings.raml))
 
-### What's included?
+### What's included
```
Comment (jstoiko) [inline annotations/security-schemes/README.md]: I think the `?` was meant to be

File: annotations/security-schemes/README.md [hunk 6]
```diff
@@ -11,14 +11,37 @@
...
+
+This directory contains the following RAML `AnnotationTypeDeclaration` fragment:
+
+- oauth-2-pkce.raml
+
+and the following example:
+
+- oauth-2-pkce-example.raml
+
+### API tools support PKCE
```
Comment (jstoiko) [inline annotations/security-schemes/README.md]: should it be `Which tools support this?` like in the section above?

File: annotations/security-schemes/README.md (hunk 5 above)
Comment (jarrodek) [reply annotations/security-schemes/README.md]: markdown linter treats it as an error.

File: annotations/security-schemes/README.md (hunk 6 above)
Comment (jarrodek) [reply annotations/security-schemes/README.md]: It's linter, again. First is the question mark and the second is a duplicated title.

File: annotations/security-schemes/README.md (hunk 5 above)
Comment (jstoiko) [reply annotations/security-schemes/README.md]: Weird. I guess that's an issue with the linter you're using...

File: annotations/security-schemes/README.md (hunk 6 above)
Comment (jstoiko) [reply annotations/security-schemes/README.md]: I don't see why markdown would forbid duplicate subtitles as long as they are under different hierarchies, which is the case here... again, I think this is an issue with the linter you're using.

**Comment by jarrodek** [comment]
@jstoiko sorry for taking so long. I corrected the titles. Should be good now.

**Review by jstoiko — APPROVED** [review]

**Pull request merged**